common/src/models/deployment.rs

```diff
@@ -7,18 +7,33 @@ use comfy_table::{
 use crossterm::style::Stylize;
 use serde::{Deserialize, Serialize};
 use std::{fmt::Display, str::FromStr};
+use ulid::Ulid;
 #[cfg(feature = "openapi")]
-use utoipa::ToSchema;
+use utoipa::{
+    openapi::{Object, ObjectBuilder},
+    ToSchema,
+};
 use uuid::Uuid;
 
+#[cfg(feature = "openapi")]
+fn ulid_type() -> Object {
+    ObjectBuilder::new()
+        .schema_type(utoipa::openapi::SchemaType::String)
+        .format(Some(utoipa::openapi::SchemaFormat::Custom(
+            "ulid".to_string(),
+        )))
+        .description(Some("String represention of an Ulid according to the spec found here: https://github.com/ulid/spec."))
+        .build()
+}
+
 #[derive(Deserialize, Serialize)]
 #[cfg_attr(feature = "openapi", derive(ToSchema))]
 #[cfg_attr(feature = "openapi", schema(as = shuttle_common::models::deployment::Response))]
 pub struct Response {
     #[cfg_attr(feature = "openapi", schema(value_type = KnownFormat::Uuid))]
     pub id: Uuid,
-    #[cfg_attr(feature = "openapi", schema(value_type = KnownFormat::Uuid))]
-    pub service_id: Uuid,
+    #[cfg_attr(feature = "openapi", schema(schema_with = ulid_type))]
+    pub service_id: Ulid,
```

Comment:
I'm wondering if we can make this backwards compatible somehow?

Reply:
Great catch. Yeh, I think newer `cargo-shuttle` clients will have trouble fetching `deployment` info based on `Uuid`, given the newer `common::deployment::Response` expects the deployer to hand over an `Ulid`. We can make this field plain, relying on the textual representation of `Uuid` and `Ulid`. This means old deployers can still serve `Uuid` (serialized they come as text and `cargo-shuttle` can use them like that), while the new deployer can serve the new `Ulid` (again, having `cargo-shuttle` display it in plain, as a regular string). Have to try it out though, but what do you think?

Reply:
Having them plain all the way sounds good :+1: